lib/hanami/application.rb

```diff
@@ -254,7 +254,8 @@ def configure_container(container)
 
           if root.join("lib").directory?
             config.component_dirs.add "lib" do |dir|
-              dir.default_namespace = application_name.to_s
+              # Register the files in `lib/[application_name]/` using top-level keys
+              dir.namespaces.add application_name.to_s, key: nil
```

Comment:
Just for the record (for those following along), we decided not to go with this `app/` dir experiment, and I've taken it a step further by making it so that there is no default dir that auto-registers into the app container.

I've just pushed https://github.com/hanami/hanami/pull/1123/commits/4177d111f198cad79fd23dfdf40fa77f74f05c02, which:

- removes that `app/` experiment
- also removes all auto-registration of components into the app container
- sets up `lib/[app_namespace]/` with zeitwerk (so classes inside the app's own namespace can be autoloaded)
- adds `lib/` to the regular old load path